serial_windows.go

```diff
@@ -178,12 +178,20 @@ func setCommState(h syscall.Handle, baud int) error {
 	return nil
 }
 
-func setCommTimeouts(h syscall.Handle) error {
+func setCommTimeouts(h syscall.Handle, readTimeout uint32) error {
 	var timeouts structTimeouts
 	const MAXDWORD = 1<<32 - 1
-	timeouts.ReadIntervalTimeout = MAXDWORD
-	timeouts.ReadTotalTimeoutMultiplier = MAXDWORD
-	timeouts.ReadTotalTimeoutConstant = MAXDWORD - 1
+	if readTimeout > 0 {
+		// non-blocking read
+		timeouts.ReadIntervalTimeout = 1000
```

Comment:
This make the inter-character timeout 1s, right?  I thought the timeout was for the overall read, not for characters?  It seems like this should be 0 instead.